packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -64,22 +64,24 @@ const Dropdown: WithComponentId<DropdownProps> = ({
 
   let dropdownTriggerer: DropdownContextType['dropdownTriggerer'];
 
-  React.Children.map(children, (child) => {
-    if (React.isValidElement(child)) {
-      if (
-        !isValidAllowedChildren(child, 'SelectInput') &&
-        !isValidAllowedChildren(child, componentIds.DropdownOverlay)
-      ) {
-        throw new Error(
-          `[Dropdown]: Dropdown can only have \`SelectInput\` and \`DropdownOverlay\` as children\n\n Check out: https://blade.razorpay.com/?path=/story/components-dropdown`,
-        );
-      }
+  React.useEffect(() => {
```

Comment:
Won't this be an issue on SSR?

Reply:
Okay reverted this, didn't cause SSR issue but it had a reason to be outside useEffect, it needs to synchronously run on component's first mount so that we can save the `dropdownTriggerer` in a variable and set it to `SelectInput` by looping over the children, if we do this in an useEffect the dependant props will cause a flashing effect.